File: subsetsTwo.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        # Sort the input to handle duplicates
        nums.sort()
        res = []

        def backtrack(start, path):
            # Add the current subset to the result
            res.append(path)
            
            # Start exploring subsets starting from the next index
            for i in range(start, len(nums)):
                # Skip duplicates
                if i > start and nums[i] == nums[i - 1]:
                    continue
                # Include current element and move to next index
                backtrack(i + 1, path + [nums[i]])

        # Start backtracking from index 0
        backtrack(0, [])
        return res

File: subsetsTwo.go
package main

import (
	"sort"
)

func subsetsWithDup(nums []int) [][]int {
	// Sort the input to handle duplicates
	sort.Ints(nums)
	var res [][]int

	var backtrack func(start int, path []int)
	backtrack = func(start int, path []int) {
		// Add the current subset to the result
		res = append(res, append([]int{}, path...))

		// Start exploring subsets from the next index
		for i := start; i < len(nums); i++ {
			// Skip duplicates
			if i > start && nums[i] == nums[i-1] {
				continue
			}
			// Include the current element and move to the next index
			backtrack(i+1, append(path, nums[i]))
		}
	}

	// Start backtracking from index 0
	backtrack(0, []int{})
	return res
}
